src/videojs-http-streaming.js

```diff
@@ -140,18 +141,56 @@ Hls.canPlaySource = function() {
     'your player\'s techOrder.');
 };
 
-const emeKeySystems = (keySystemOptions, videoPlaylist, audioPlaylist) => {
+const emeKeySystems = (keySystemOptions, mainSegmentLoader, audioSegmentLoader) => {
   if (!keySystemOptions) {
     return keySystemOptions;
   }
 
+  let videoMimeAndCodec;
+  let audioMimeAndCodec;
+
+  // if there is a mimeType associated with the audioSegmentLoader, then the audio
+  // and video mimeType and codec strings are already in the format we need to
+  // pass with the other key systems
+  if (audioSegmentLoader.mimeType_) {
+    videoMimeAndCodec = mainSegmentLoader.mimeType_;
+    audioMimeAndCodec = audioSegmentLoader.mimeType_;
+
+  // if there is no audioSegmentLoader mimeType, then we have to create the
+  // the audio and video mimeType/codec strings from information extrapolated
+  // from the mainSegmentLoader mimeType (ex. 'video/mp4; codecs="mp4, avc1"' -->
+  // 'video/mp4; codecs="avc1"' and 'audio/mp4; codecs="mp4"')
+  } else {
+    const splitMime = mainSegmentLoader.mimeType_.split(';');
```

Comment:
Would it make sense to put this whole block of code in `src/utils/codecs`? something like `parseMimeType` that returns something like: `{video: {type: 'mp4', codecs: ['mp4, 'avc1']`?

Reply:
How about using `parseContentType()` from the codec utils which already does almost the same thing?